tests/test_transformers.py

```diff
@@ -120,6 +120,71 @@ def test_backcomp(self):
             'test f1 = {}\nLOG:\n{}'.format(test['f1'], stdout)
         )
 
+    @testing_utils.retry(ntries=3)
+    def test_xlm(self):
+        stdout, valid, test = testing_utils.train_model(dict(
+            task='integration_tests:candidate',
+            model='transformer/ranker',
+            optimizer='adamax',
+            learningrate=7e-3,
+            batchsize=16,
+            validation_every_n_epochs=5,
+            validation_patience=2,
+            n_layers=1,
+            n_heads=4,
+            ffn_size=64,
+            embedding_size=32,
+            candidates='batch',
+            eval_candidates='inline',
+            gradient_clip=0.5,
+            variant='xlm',
+            activation='gelu',
+        ))
+
+        self.assertGreaterEqual(
+            valid['hits@1'],
+            0.90,
+            "valid hits@1 = {}\nLOG:\n{}".format(valid['hits@1'], stdout)
+        )
+        self.assertGreaterEqual(
+            test['hits@1'],
+            0.90,
+            "test hits@1 = {}\nLOG:\n{}".format(test['hits@1'], stdout)
+        )
+
+    @testing_utils.retry(ntries=3)
+    def test_alt_reduction(self):
```

Comment:
test name is a bit unclear if we're specifically testing reduction_type == 'first'

Reply:
It was meant to be a test that just tries something /other/ than `mean`. `max` we know doesn't work that great.